source/speech/delayedCharacterDescriptions.py

```diff
@@ -42,29 +39,27 @@ def cancelDelayedCharacterDescription() -> None:
 
 def _speakDelayedCharacterDescription(info: _DelayedCharacterDescriptionTextInfo) -> None:
 	"""
-	this is the function that will be called from the timer after N milliseconds.
-	This function will check if a character description is available and then,
-	speak it by using "spellTextInfo(..., useCharacterDescriptions=True)"
+	Should be called from a timer after a delay.
+	Will check if a character description is available for a locale, then speak it.
 	"""
 	if info.text.strip() == "":
 		return
-	from .speech import getCharDescListFromText, spellTextInfo, getCurrentLanguage
+	from .speech import getCharDescListFromText, _spellTextWithFields, getCurrentLanguage
 	curLang = getCurrentLanguage()
 	if config.conf['speech']['autoLanguageSwitching']:
-		for k in info.fields:
-			if isinstance(k, textInfos.FieldCommand) and k.command == "formatChange":
-				curLang = k.field.get('language', curLang)
+		for command in info.getTextWithFields():
+			if isinstance(command, textInfos.FieldCommand) and command.command == "formatChange":
+				curLang = command.field.get('language', curLang)
 	_, description = getCharDescListFromText(info.text, locale=curLang)[0]
-	# We can't call spellTextInfo directly because we need to check if the description is available first.
 	if description:
-		spellTextInfo(info, useCharacterDescriptions=True)
+		_spellTextWithFields(info, useCharacterDescriptions=True)
```

Comment:
if `spellTextInfo` is ever updated to use other aspects of a `TextInfo`, calling it with a `_DelayedCharacterDescriptionTextInfo` may cause an error. As such, I have split out `_spellTextWithFields` from `spellTextInfo`. `_spellTextWithFields` supports the shared Protocol `_SupportsGetTextWithFields` used by both `TextInfo` and `_DelayedCharacterDescriptionTextInfo`.